CMakeLists.txt

```
@@ -1,7 +1,8 @@
-cmake_minimum_required(VERSION 3.15)
+cmake_minimum_required(VERSION 3.15...3.21)
```

for[UNITY_BUILD]( https://cmake.org/cmake/help/latest/prop_tgt/UNITY_BUILD.html) it should be `3.19...3.21`

---

I am unclear on the purpose of the [`<policy_max>`](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html) parameter here. 

Broadly speaking, I envision two kinds of people using this project:
1. Users like yourself who know CMake well enough to know the exact range of CMake versions that will work for their project. These users will know about the `<policy_max>` parameter, and will set it to the appropriate value for themselves.
2. Users who don't know so much about CMake yet, and will be very frustrated when they try to compile this project for the first time, because their package manager has installed CMake 3.22.

I do not understand how setting `<policy_max>` in the starter project is of benefit to either user. Can you please provide some more context here?

---

The max version is the highest policy used be `cmake` if a newer version of `cmake` is used.

I have tested with `3.21`, so I can guaranty this `cmake build system` work with it.
but you need at leased `3.16` to use the `UNITY_BUILD` property!

---

```suggestion
cmake_minimum_required(VERSION 3.16)
```
Please remove the `<policy_max>` parameter. Unless you have a good reason to use it, it will break our users' code when they upgrade CMake. You don't have to guarantee that your change will work with future versions of CMake.

---

> ...  it will break our users' code when they upgrade `CMake`. You don't have to guarantee that your change will work with future versions of CMake.

That is simply not true!

I can guaranty it for `CMake 3.21.3` And `CMake ` does it with this concept: https://cmake.org/cmake/help/v3.21/manual/cmake-policies.7.html?highlight=policy

---

> > ...  it will break our users' code when they upgrade `CMake`. You don't have to guarantee that your change will work with future versions of CMake.
> 
> That is simply not true!

You are correct; this assessment was premature. `<policy_max>` is intended to ensure forwards compatibility with future versions of CMake, not to break it. This code will ensure that any user who configures the project with CMake 3.22+ will have the same experience as someone using CMake 3.21.

The part that I still don't understand is why you have selected 3.21 as the `<policy max>`. If it's just because 3.21 is the newest CMake currently available, then I think this could be the wrong decision. Will we need to bump this version number every time a new CMake is released? I don't want to review an endless stream of PRs bumping that version number for no benefit.

Unless I am misunderstanding this, setting `policy_max` to 3.21 implies that 6 different policies might be used (policies 3.16 through 3.21), depending on what version of CMake you are using. I think it would make sense to set `minimum_required` and `policy_max` to the same number to eliminate this ambiguity.

Someone using this project could add some code that depends on the policy for CMake 3.20 without realizing it; they will be in for a nasty surprise when someone else builds the project with CMake 3.19 and gets a different behavior. If both `minimum_required` and `policy_max` were set to 3.16, this kind of thing could not happen.

Additionally, I think it's worth adding a comment to describe what's happening here. Perhaps I am the only person in the world who will misunderstand how `policy_max` works, but somehow I doubt it.